test/inductor/test_binary_folding.py

```diff
@@ -232,6 +209,99 @@ def my_inner_compile(gm, example_inputs, *args, **kwargs):
+                    super().__init__()
+                    self.linear = nn.Linear(
+                        in_channels, out_channels, bias=use_bias, **kwargs
+                    ).to(device)
+                    self.use_scalar = scalar
+                    tensor_size = [
+                        self.linear.weight.size(0),
+                    ]
+                    self.tensor = torch.nn.Parameter(
+                        add_tensor
+                        if add_tensor is not None
+                        else torch.rand(tensor_size).to(device)
+                    )
+                    self.op = op
+
+                def forward(self, x):
+                    x = self.linear(x)
+                    if self.use_scalar:
+                        return self.op(x, 2.0)
+                    else:
+                        return self.op(x, self.tensor)
+
+            torch._dynamo.reset()
+            counters.clear()
+            mod_eager = LinearOp(3, 32, self.device).eval()
+            out_optimized = torch.compile(mod_eager)
+
+            torch.manual_seed(1234)
+            inp = torch.rand([4, 3]).to(self.device)
+            out_eager = mod_eager(inp)
+            out_optimized = out_optimized(inp)
+            self.assertEqual(out_optimized, out_eager, atol=5e-05, rtol=5e-06)
+            if expect_success:
+                self.assertEqual(counters["inductor"]["binary_folding"], 1)
+            else:
+                self.assertEqual(counters["inductor"]["binary_folding"], 0)
+
+        linear_bias = [True, False]
+        use_scalar = [True, False]
+        ops = [torch.add, torch.sub, torch.mul, torch.div]
+        add_tensor_size = [
+            [
+                32,
+            ],
+            [1, 32],
+            [
+                1,
+            ],
+            [1, 1],
+        ]
+        for use_bias, pytorch_op, scalar, tensor_size in itertools.product(
+            linear_bias, ops, use_scalar, add_tensor_size
+        ):
+            test_linear_fusion(
+                use_bias,
+                pytorch_op,
+                scalar,
+                add_tensor=torch.rand(tensor_size).to(self.device),
+                expect_success=True,
+            )
+
+        for use_bias, pytorch_op in itertools.product(linear_bias, ops):
+            # add
+            test_linear_fusion(
+                use_bias,
+                pytorch_op,
+                False,
+                add_tensor=torch.rand(
+                    4,
+                    32,
+                ).to(self.device),
+                expect_success=False,
+            )
+
+            # broadcasting add
+            test_linear_fusion(
+                use_bias,
+                pytorch_op,
+                False,
+                add_tensor=torch.rand(
+                    4,
+                    1,
+                ).to(self.device),
+                expect_success=False,
+            )
```

Comment:
Please add note in the code on why you are expecting fusion failures.

Reply:
Added, thanks for your comments!